core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -373,6 +374,22 @@ object SparkSubmit {
       sysProps("spark.submit.pyArchives") = pyArchives
     }
 
+    // In yarn mode for an R app, add the SparkR package archive to archives
+    // that can be distributed with the job
+    if (args.isR && clusterManager == YARN) {
+      val sparkHome = sys.env.get("SPARK_HOME")
+      if (sparkHome.isEmpty) {
+        printErrorAndExit("SPARK_HOME does not exist for R application in yarn mode.")
+      }
+      val rPackagePath = Seq(sparkHome.get, "R", "lib").mkString(File.separator)
+      val rPackageFile = new File(rPackagePath, SPARKR_PACKAGE_ARCHIVE)
+      if (!rPackageFile.exists()) {
+        printErrorAndExit(s"$SPARKR_PACKAGE_ARCHIVE does not exist for R application in yarn mode.")
+      }
+      val localURI = Utils.resolveURI(rPackageFile.getAbsolutePath)
+      args.archives = mergeFileLists(args.archives, localURI.toString + "#sparkr")
```

Comment:
Could you add a comment here as to why we have this '#sparkr" ? I believe this is to get the archive to unzip to a symlink named `sparkr` ?